### DISCUSSION TITLE
Season is incorrectly set for certain games

### DISCUSSION THREAD

**Discussion by adamrusko** [dcaribou/transfermarkt-datasets#282] [Q&A] [answered]
Hi guys,
I've started to use this dataset for my thesis. And at one point I've seen quite a big problem which could be solved if we could produce a good solution for the problem which is:
I made two prompts where the user can select a league (like Bundesliga or LaLiga) and certain season (like 2015 or 2021). After they've selected this two values, I've declared a query builder to list all the matches from the chosen league in the chosen season. And this is point where the problem comes in.
*now you should take a look at the photo*
I want to tell you that not all the leagues have this kind of problem. I've seen this problem at the Ukranian (Premier League) and the Belgian league (Jupiler Pro League). The picture represents all the league matches in the Belgian Jupiler Pro League in 2021. The problem is that although the chosen season was 2021, you can see records where the date is July 2022 and August 2022, these records should not be here because these records should be there for season 2022 not 2021 and also there are missing records from the games which were played in July 2021 and August 2021. As I said before, this is not a problem for every league, and even if this problems happen in certain leagues it is only problems in the first few rounds/matchdays and then all the matchdays are working properly.
![kép_2024-02-16_113912593](https://github.com/dcaribou/transfermarkt-datasets/assets/78847191/f1aa1f60-e92b-405d-8e8e-b9964a01be75)
![belgian league 2021_ round 1-2](https://github.com/dcaribou/transfermarkt-datasets/assets/78847191/17edf73f-68bf-4c8e-bd3f-a6812c99e279)
![belgian league 2021_ round 3-4](https://github.com/dcaribou/transfermarkt-datasets/assets/78847191/34f69ef3-8ecd-4ad2-b597-e94ac2db87e6)

**Comment by dcaribou** [comment]
I @adamrusko,
I am looking at BE1 league data on my side rn. I looked at a couple of games (`3848871` and `3848874`) that show in your screenshot and they are correctly labeled as belonging to season 2022. Maybe you are using an outdated version of the data?

![image](https://github.com/dcaribou/transfermarkt-datasets/assets/16071835/4950e0c5-a9dc-45d4-baa8-d60080610582)

As for the missing games you mention, could you provide an example?

Btw, I am going to update the title of your discussion to something a little more descriptive of the issue you are reporting. I hope you don't mind.

**Comment by adamrusko** [comment]
I checked all the data again, downloaded the newly updated tables from data.world and the problem is still there. This time I was using notepad and Excel to search for the record where the game_id was 3848871 in the table labelled Games. And the season is still 2021 instead of 2022.

**Reply by dcaribou** [reply, marked as answer]
That's strange. I've tried it once again from my side, and I see season 2022 when I download the data from data.world...
I'd suggest that you get the data [from Kaggle](https://www.kaggle.com/datasets/davidcariboo/player-scores) then, and try again.

<img width="1023" alt="Screenshot 2024-02-19 at 16 21 47" src="https://github.com/dcaribou/transfermarkt-datasets/assets/16071835/a8351f6d-c405-43fa-abc3-4b5a15e7c98c">
